test-framework/core/src/main/java/org/keycloak/testframework/https/ManagedCertificates.java

```diff
@@ -30,31 +30,35 @@ public class ManagedCertificates {
 
     private KeyStore serverKeyStore;
     private KeyStore clientsTrustStore;
+    private final Path serverKeystorePath;
+    private final Path clientsTruststorePath;
+    private final char[] password;
 
     private final static Path KEYSTORES_DIR = Path.of(System.getProperty("java.io.tmpdir"));
-    private final static Path SERVER_KEYSTORE_FILE_PATH = KEYSTORES_DIR.resolve("kc-testing-server-keystore.jks");
-    private final static Path CLIENTS_TRUSTSTORE_FILE_PATH = KEYSTORES_DIR.resolve("kc-testing-clients-truststore.jks");
-
-    private final static char[] PASSWORD = "password".toCharArray();
-
     private final static String PRV_KEY_ENTRY = "prvKey";
     public final static String CERT_ENTRY = "cert";
 
 
-    public ManagedCertificates() throws ManagedCertificatesException {
+    public ManagedCertificates(CertificatesConfigBuilder configBuilder) throws ManagedCertificatesException {
         if (!CryptoIntegration.isInitialised()) {
             CryptoIntegration.setProvider(new DefaultCryptoProvider());
         }
         cryptoProvider = CryptoIntegration.getProvider();
-        initServerCerts();
+
+        serverKeystorePath = KEYSTORES_DIR.resolve("kc-testing-server-keystore" + "." + configBuilder.getKeystoreFormat().getPrimaryExtension());
+        clientsTruststorePath = KEYSTORES_DIR.resolve("kc-testing-clients-truststore" + "." + configBuilder.getKeystoreFormat().getPrimaryExtension());
+
+        password = configBuilder.getKeystoreFormat() == KeystoreUtil.KeystoreFormat.JKS ? "password".toCharArray() : "passwordpassword".toCharArray();
```

Comment:
I don't find this necessary. The password could be `"passwordpassword"` in both cases if it's needed for FIPS.